File: C++/solve.cpp
#include <iostream>
#include <vector>
#include <chrono> // Para medir el tiempo de ejecución

// Función que implementa el algoritmo de ordenación Burbuja
void burbuja(std::vector<int>& matriz) {
    int n = matriz.size();
    for (int i = 0; i < n - 1; ++i) {
        // Si no hubo intercambio, el arreglo ya está ordenado
        bool intercambiado = false;
        
        for (int j = 0; j < n - i - 1; ++j) {
            if (matriz[j] > matriz[j + 1]) {
                // Intercambiamos los elementos
                std::swap(matriz[j], matriz[j + 1]);
                intercambiado = true;
            }
        }

        // Si no hubo intercambio, el arreglo ya está ordenado
        if (!intercambiado) {
            break;
        }
    }
}

// Función para imprimir la matriz
void imprimirMatriz(const std::vector<int>& matriz) {
    for (int i = 0; i < matriz.size(); ++i) {
        std::cout << matriz[i] << " ";
    }
    std::cout << std::endl;
}

int main() {
    // Creamos una matriz desordenada
    std::vector<int> matriz = {64, 25, 12, 22, 11};

    std::cout << "Matriz original: ";
    imprimirMatriz(matriz);

    // Iniciamos el cronómetro
    auto inicio = std::chrono::high_resolution_clock::now();

    // Llamamos a la función de ordenación
    burbuja(matriz);

    // Detenemos el cronómetro
    auto fin = std::chrono::high_resolution_clock::now();

    std::cout << "Matriz ordenada: ";
    imprimirMatriz(matriz);

    // Calculamos y mostramos el tiempo de ejecución
    std::chrono::duration<double> tiempo = fin - inicio;
    std::cout << "Tiempo de ejecución: " << tiempo.count() << " segundos." << std::endl;

    return 0;
}

File: Java/solve.java
public class Solve {

    public static void ordenarArreglo(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - 1 - i; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }
    public static void imprimirArreglo(int[] arr) {
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        int[] arr = {64, 25, 12, 22, 11};
        long inicio = System.nanoTime();
        System.out.println("Arreglo original:");
        imprimirArreglo(arr);
        ordenarArreglo(arr);

        // Mide el tiempo de ejecución después de la ordenación
        long fin = System.nanoTime();
        System.out.println("\nArreglo ordenado:");
        imprimirArreglo(arr);

        // Calcula y muestra el tiempo de ejecución
        long tiempoEjecucion = fin - inicio;
        System.out.println("\nTiempo de ejecución: " + tiempoEjecucion + " nanosegundos");
    }
}
